Test program to demostrate OMP_PLACES to control affinity.

raghavendhra opened the pull request.

ronlieb commented:
please move the test case to smoke-fails , also  this patch seems to also contain the hwloc implementation which belongs in a separate patch.

raghavendhra commented:
> please move the test case to smoke-fails , also this patch seems to also contain the hwloc implementation which belongs in a separate patch.

Done! Please check.

ronlieb reviewed (changes requested):
the patch has the hwloc changes and the test case.
The title says test case.
two patches i would think.

raghavendhra committed:
Add smoke test to control affinity using omp_places.
	modified:   Makefile
	new file:   omp_places/Makefile
	new file:   omp_places/omp_places.c

raghavendhra commented:
> the patch has the hwloc changes and the test case. The title says test case. two patches i would think.

I removed hwloc patch for this PR. Probably Greg will push that patch with his PR. 
Also I removed numa_domains test cases for now and will wait for Greg's hwloc patch to land and then add OMP_PLACES="numa_domains(1)" test cases.
